webpack/assets/javascripts/react_app/common/HOC.js

```diff
@@ -20,22 +20,21 @@ export const callOnMount = callback => WrappedComponent => componentProps => {
  * assuming the component has withRouter
  * @param {Function} callback - function to run
  */
-export const callOnPopState = callback => WrappedComponent => componentProps => {
+export const callOnPopState = callback => WrappedComponent => props => {
   const didMount = useRef(false);
-
+  const {
+    history: { action },
+    location: { search },
+  } = props;
   useEffect(() => {
-    const {
-      history: { action },
-    } = componentProps;
-
     if (action === 'POP' && didMount.current) {
-      callback(componentProps);
+      callback(props);
     } else {
       didMount.current = true;
     }
-  }, [componentProps.location.search]);
+  }, [search, action, props]);
```

Comment:
@tbrisker @mmoll, this is more than a linting change

Reply:
good point, i guess this is what is causing the test failures.